allure-app-under-test/src/main/java/com/capgemini/ntc/selenium/pages/projectY/DisappearingElementsPage.java

```diff
@@ -0,0 +1,73 @@
+package com.capgemini.ntc.selenium.pages.projectY;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import com.capgemini.ntc.selenium.core.BasePage;
+import com.capgemini.ntc.test.core.logger.BFLogger;
+
+public class DisappearingElementsPage extends BasePage {
+	
+	private static final By	selectorGalleryMenuButton	= By.cssSelector("li > a[href*=gallery]");
+	private static final By	selectorMenuButtons			= By.cssSelector("li");
+	
+	@Override
+	public boolean isLoaded() {
+		getDriver().waitForPageLoaded();
+		return getDriver().getCurrentUrl()
+						.contains("disappearing_elements");
+	}
+	
+	@Override
+	public void load() {
+		BFLogger.logDebug("load()");
+	}
+	
+	@Override
+	public String pageTitle() {
+		return "The Internet";
+	}
+	
+	/**
+	 * Returns a number of WebElements representing menu buttons.
+	 * 
+	 * @return A number of WebElements.
+	 */
+	public int getNumberOfMenuButtons() {
+		return getDriver().findElementDynamics(selectorMenuButtons)
+						.size();
+	}
+	
+	/**
+	 * Returns WebElement representing disappearing element of menu.
+	 * 
+	 * @return Disappearing WebElement if visible, null otherwise.
+	 */
+	public WebElement getGalleryMenuElement() {
+		return getDriver().findElementQuietly(selectorGalleryMenuButton);
+	}
+	
+	/**
+	 * Refreshes web page as many times as it is required to appear/disappear menu button WebElement.
+	 * 
+	 * @param appearance
+	 *            Determines if element should appear (true) or disappear (false).
+	 */
+	public void refreshUntilAppear(boolean appearance) {
+		int numberOfAttempts = 5;
+		int counter = 0;
+		while (!(!(isGalleryMenuElementVisible() ^ appearance)) || counter > numberOfAttempts) {
```

Comment:
that is tough one. 
`(!(!(isGalleryMenuElementVisible() ^ appearance)) || counter > numberOfAttempts)`
Could you please split this into some self descriptive private variable/method